Assets/Scripts/Enemy Ai/AttackState.cs

```diff
@@ -16,7 +16,11 @@ public class AttackState : State
     void Start()
     {
         //player = GameObject.Find("TopDownCharacter");
+    }
 
+    public override string GetStateAsString()
```

Comment:
This looks really bad. Override implies that this is a virtual function call that incurs 3 cache misses, and it is very bad for performance. Additionally, we should try to avoid introducing a string period. There's no reason to use a dynamically sized char array to identify a state. Ideally, we would use an Enum or typeof.

Reply:
Technically strings in c# are immutable and statically sized. Using strings to identify the type of the state is bad though.